server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -174,6 +174,19 @@ public RepositoryData addSnapshot(final SnapshotId snapshotId,
         return new RepositoryData(genId, snapshots, newSnapshotStates, allIndexSnapshots, incompatibleSnapshotIds);
     }
 
+    /**
+     * Create a new instance with the given generation and all other fields equal to this instance.
+     *
+     * @param newGeneration New Generation
+     * @return New instance
+     */
+    public RepositoryData withGenId(long newGeneration) {
```

Comment:
Had to add this for the test, but I think it can be used in a follow up to clean up handling of the generation on repository data instances a little as well (currently the generation on these instances doesn't always correspond the the actual generation it will be written as).